langchain/agents/agent_toolkits/openapi/planner.py

```diff
@@ -111,15 +113,25 @@ def _create_api_planner_tool(
     return tool
 
 
+def _get_llm_chain(llm):
+    if isinstance(llm, AzureOpenAI) or isinstance(llm, AzureChatOpenAI):
+        llm_args = llm._identifying_params.copy()
+        llm_args.pop("model_name")
+        llm = AzureOpenAI(**llm_args)
```

Comment:
Question - why do we need to instantiate a new `AzureOpenAI` instance here?

Reply:
Thanks for your review. `RequestsGetToolWithParsing` creates a new `OpenAI` with the default `model_name` on instantiation (https://github.com/hwchase17/langchain/issues/2546#issuecomment-1505136535), so I'm just recreating that same behavior but with the correct class when an `Azure` llm is being used. That llm instance will have the `engine` parameter, which is necessary to call the OpenAI API (as it is now, the call fails).

>we shouldn't be handling individual LLM providers in this file

The original code is creating instances of `OpenAI` so it already handles specific providers. I wonder then if it's necessary to do some additional refactoring, so that particular providers don't show at all here.

Reply:
That's a good solution. I made those changes and pushed them, and can confirm that this works with Azure as well.
Please let me know if I should make any other change.

Reply:
@vowelparrot please let me know if this is ok